src/Microsoft.AspNetCore.Mvc.Core/Internal/FilterActionInvoker.cs

```diff
@@ -294,9 +304,12 @@ private Task InvokeAllResourceFiltersAsync()
                     }
                     else
                     {
+                        await InvokeResourceFilterAsync();
+                        Debug.Assert(_resourceExecutedContext != null);
+
                         _diagnosticSource.BeforeOnResourceExecuted(_resourceExecutedContext, item.Filter);
 
-                        item.Filter.OnResourceExecuted(await InvokeResourceFilterAsync());
+                        item.Filter.OnResourceExecuted(_resourceExecutedContext);
```

Comment:
Discussed in person and we're going to add asserts to all of these methods. There's an extremely high cost/benefit ratio to adding unit tests for all of these diagnostic source methods.